### PR TITLE
Multilang support for content create/update

### PR DESCRIPTION
**Pull request by damianz5**
Multilang support for content create/update

todo:
- [x] add test
- [x] check travis result

### CODE REVIEW

**Comment by damianz5** [comment]
ping @gggeek please review

**Comment by SylvainGuittard** [comment]
ping @gggeek ;)

**Comment by gggeek** [comment]
Sorry for taking a while. PR reviewed, changes requested.

File: Core/Executor/ContentManager.php [hunk 1]
```diff
@@ -599,29 +599,70 @@ protected function setFields($createOrUpdateStruct, array $fields, ContentType $
...
+                $fieldDefinition = $contentType->fieldDefinitionsByIdentifier[$fieldIdentifier];
+                $fieldValue = $this->getFieldValue($fieldValue, $fieldDefinition, $contentType->identifier, $step->context);
+                $createOrUpdateStruct->setField($fieldIdentifier, $fieldValue, $language);
+            }
+            $i++;
+        }
+    }
 
-            $createOrUpdateStruct->setField($fieldIdentifier, $fieldValue, $this->getLanguageCode($step));
+    protected function hasLanguageCodesAsKeys(array &$array)
```
[the remark this answers is not in the document]
Comment (damianz5) [reply Core/Executor/ContentManager.php]: Updated, please review again

File: Core/Executor/ContentManager.php [hunk 2]
```diff
@@ -595,33 +595,137 @@ public function generateMigration(array $matchCondition, $mode, array $context =
...
+     * Checks whatever at least one of field is using multilang syntax.
+     *
+     * @param array $fields
+     * @return bool
+     */
+    protected function hasLanguageCodesAsKeys(array $fields)
+    {
+        $languages = $this->getContentLanguages();
+        $languageCodes = array_combine($languages, $languages);
+        $hasLanguageCodesAsKeys = false;
```
Comment (gggeek) [inline Core/Executor/ContentManager.php]: I'd rather do the opposite logic: start the function with True and exit with False as soon as one non-multilang field def (or multilang with wrong key) is detected.
Will merge as is, then fixup

File: Core/Executor/ContentManager.php [hunk 3]
```diff
@@ -595,33 +595,137 @@ public function generateMigration(array $matchCondition, $mode, array $context =
...
+    }
+
+    /**
+     * Returns all Languages.
+     *
+     * @return array
+     */
+    protected function getContentLanguages()
+    {
+        static $languages;
```
Comment (gggeek) [inline Core/Executor/ContentManager.php]: I think that this (static) might not be 100% safe, as there is a contrived case where we could access many repos at the same time

**Pull request merged**